typing/typecore.ml

```diff
@@ -462,7 +447,15 @@ let has_variants p =
 
 (* pattern environment *)
 type pattern_variable =
-  Ident.t * type_expr * string loc * Location.t * bool (* as-variable *)
+  {
+    pv_id: Ident.t;
+    pv_type: type_expr;
+    pv_name: string loc;
```

Comment:
I believe the `pv_name` field is useless. Not only is it redundant with the `pv_id` and `pv_loc` fields, as far as I can tell it's only "used" to be stuffed in `Tcl_fun`, and then completely ignored when `Tcl_fun` is processed in `translclass.ml`.

This remark is somewhat orthogonal to your PR, but since you're cleaning up that area, might as well go all the way :)
Do you mind trying to add a commit which removes that field?

Reply:
This is done, I removed the corresponding field from both `Tcl_fun` and `Tcl_let`, which resulted in a nice cleanup.